apps/mobile/src/libs/components/text/text.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { ReactNode } from "react";
+import { Text as RNText, type StyleProp, type TextStyle } from "react-native";
+
+import {
+	BaseColor,
+	FontFamilies,
+	FontSize,
+	LineHeight,
+	WeightMap,
+} from "~/libs/enums/enums";
+
+import { styles as presetStyles } from "./styles";
+
+type Properties = {
+	children: ReactNode;
+	color?: (typeof BaseColor)[keyof typeof BaseColor];
+	fontFamily?: keyof typeof FontFamilies;
+	preset?: "default" | "heading" | "subheading";
+	size?: keyof typeof FontSize;
```

Comment:
Use `ValueOf<FontSize>` instead

Reply:
Before I received comment:
<img width="822" alt="Screenshot 2024-08-19 at 13 51 30" src="https://github.com/user-attachments/assets/c1e52437-8744-43dc-9bc7-7c6539bd40b6"> and that's why size is keyof typeof FontSize;

Reply:
I believe it would be better to send enum value in `Properties` and make a default one `FontSize.MD`